kazoo/protocol/connection.py

```diff
@@ -380,7 +381,8 @@ def _read_response(self, header, buffer, offset):
             # Determine if watchers should be registered
             watcher = getattr(request, 'watcher', None)
             if not client._stopped.is_set() and watcher:
-                if isinstance(request, GetChildren):
+                if (isinstance(request, GetChildren)
+                    or isinstance(request, GetChildren2)):
```

Comment:
@saffroy FYI `isinstance()` can take multiple values as a tuple, so that'd be the easiest way to clean this up:

    if isinstance(request, (GetChildren, GetChildren2)):

Details: https://stackoverflow.com/a/33311330/770425

Reply:
Ah, that should do. And I learned something!